### PR TITLE
Multi-attribute index support

### PR DESCRIPTION
**Pull request by tywalch**
# Context 
AWS has added [support for multi-attribute indexes](https://aws.amazon.com/blogs/database/multi-key-support-for-global-secondary-index-in-amazon-dynamodb/) in DynamoDB. Support for this new functionality would be immensely helpful for users and is likely possible without breaking changes.

# Approach
In this PR, I will be chipping away at support for this feature. My priority is to just-get-things-working™ _with_ tests. Ideally, this uncovers edge cases and tough judgment calls. 

Only once things are working and behavior is locked down, I'll revisit ways to improve code organization, error handling, documentation, etc.

### CODE REVIEW

**Comment by anatolzak** [comment]
Hey @tywalch! Fortunately, AWS released an update yesterday to DynamoDB Local 3.3.0 that adds support for multi-attribute GSIs, so we no longer need to use an AWS-hosted DynamoDB table to test the new GSI 🎉

**Comment by tywalch** [comment]
@anatolzak I believe this feature may be finished. Would you mind taking a once-over look to see if anything jumps out to you?

**Comment by anatolzak** [comment]
> @anatolzak I believe this feature may be finished. Would you mind taking a once-over look to see if anything jumps out to you?

@tywalch of course! I will take a look tomorrow (it's night time where I am)

**Comment by anatolzak** [comment]
@tywalch I briefly reviewed the PR, but primarily conducted manual testing and found a [few issues](https://github.com/tywalch/electrodb/pull/537#pullrequestreview-3842008907). 

Let me know when you'd like me to take another look. I will be happy to do so!

File: www/src/pages/en/modeling/indexes.mdx [hunk 1]
```diff
@@ -743,3 +747,369 @@ When returned from a query, however, ElectroDB will return the following and tri
...
+}
+```
+
+### Querying Multi-Attribute Indexes
+
+In November 2025, DynamoDB added support for querying [multi-attribute GSIs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.DesignPattern.MultiAttributeKeys.html) using the `KeyConditionExpression` syntax. ElectroDB's `composite` index type is designed to take full advantage of this feature, allowing you to query each composite attribute directly without needing to concatenate values into a single key.
+
+Querying a `composite` index works the same as other ElectroDB index types. You provide pk composite attributes to initiate the query, and can use sort key operations like `begins`, `between`, `gt`, `gte`, `lt`, and `lte` on the sk composite attributes.
+
+The key difference is in the generated DynamoDB parameters: instead of a single `KeyConditionExpression` using `begins_with` on a concatenated sort key, ElectroDB generates individual equality and comparison conditions for each attribute.
```
Comment (tywalch) [inline www/src/pages/en/modeling/indexes.mdx]:
```suggestion
```

File: www/src/pages/en/modeling/indexes.mdx [hunk 2]
```diff
@@ -743,3 +747,369 @@ When returned from a query, however, ElectroDB will return the following and tri
...
+
+2. **No `field` property**: Composite indexes must not include a `field` property on `pk` or `sk`. The attribute's own `field` mapping (defined in the `attributes` section) determines the DynamoDB column name. Including a `field` property on a composite index will result in an error.
+
+3. **No `condition` callback**: Composite indexes do not support the `condition` callback used for [sparse indexes](#sparse-indexes). Including a `condition` on a composite index will result in an error.
+
+4. **No `template` property**: Since composite indexes store attribute values directly (without concatenation or formatting), the `template` property is not applicable.
+
+5. **No collections**: `composite` indexes do not support collections because these indexes use attribute values directly.
+
+### Gotchas and Tips
```
Comment (tywalch) [inline www/src/pages/en/modeling/indexes.mdx]:
```suggestion
### How comparison works with composite indexes
For composite indexes, comparison operators (`gt`, `gte`, `lt`, `lte`, `between`) apply only to the last attribute you provide in the query.

Think of a composite index like a hierarchy of folders:

- The first attribute narrows you to a top-level folder.
- The second attribute narrows you further.
- The third attribute narrows you even further.
- And so on.

You must fully specify each level using equality (`=`) until you reach the last level you provide.
Only at that final level can you use a range comparison (`gt`, `lt`, etc.).

This is different from clustered or isolated indexes.
- clustered and isolated indexes first concatenate all attributes into a single string (left to right).
- Comparisons are then applied to that one combined string.
- As a result, the comparison operates on the entire concatenated value — not on individual attributes.

By contrast, composite indexes treat each attribute separately and rely on DynamoDB’s rule:
**Range comparisons can only be applied to the final attribute in the sort key condition.**

If your index has only one attribute, both approaches mostly behave the same. The difference appears when multiple attributes are involved.

### Gotchas and Tips
```
